### PR TITLE
Add view_concat for edge_minor_property_view_t and update transform_reduce_e_by_dst_key to support reduce_op on tuple types

### LINKED ISSUE #2878: [FEA]: Add view_concat for edge_minor_property_view_t  and update transform_reduce_e_by_dst_key to reduce tuple types
### Is this a new feature, an improvement, or a change to existing functionality?

Improvement

### How would you describe the priority of this feature request

High

### Please provide a clear description of problem this feature solves

For Leiden implementation, we need to concatenate multiple edge_minor_property_view_t  and  support reduce ops on tuple type using transform_reduce_e_by_dst_key 

### Describe your ideal solution

All partition belonging to a GPU have same range for edge_minor_property_view_t  and hence simpler to deal with. 

### Describe any alternatives you have considered

_No response_

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow cuGraph's Code of Conduct
- [X] I have searched the [open feature requests](https://github.com/rapidsai/cugraph/issues?q=is%3Aopen+is%3Aissue+label%3A%22feature+request%22%2Cimprovement%2Cenhancement) and have found no duplicates for this feature request

https://github.com/rapidsai/cugraph/issues/2878#tasklist-block-463ff927-bb54-43bc-b7a0-37e9b800ca85

### PR DESCRIPTION
**Pull request by naimnv**
- Add view_concat for edge_minor_property_view_t 
- update transform_reduce_e_by_dst_key to support reduce_ops on tuple types
closes #2878

### CODE REVIEW

File: cpp/src/prims/transform_reduce_e_by_src_dst_key.cuh [hunk 1]
```diff
@@ -66,7 +66,7 @@ __device__ void update_buffer_element(
   EdgePartitionSrcDstKeyInputWrapper edge_partition_src_dst_key_input,
   EdgeOp e_op,
   typename GraphViewType::vertex_type* key,
-  T* value)
+  ValueIterator value_first)
```
Comment (seunghwak) [inline cpp/src/prims/transform_reduce_e_by_src_dst_key.cuh:69]: A bit of nitpicking,

but in our naming convention, `value_first` implies the first element of a multi-element array. But here, `value_first` is actually a reference to the one buffer element we need to update. So, better rename this to something like just  `value` or `value_iter`.

**Review by seunghwak — COMMENTED** [review]
Looks good to me except for some minor cosmetic issues.

Have you tried to compile this?

**Comment by seunghwak** [comment]
And I assumed this is ready-for-review.

**Review by seunghwak — APPROVED** [review]

File: cpp/src/prims/transform_reduce_e_by_src_dst_key.cuh (hunk 1 above)
Comment (naimnv) [reply cpp/src/prims/transform_reduce_e_by_src_dst_key.cuh]: Renamed.

**Review by ChuckHastings — APPROVED** [review]

**Review by jnke2016 — APPROVED** [review]

**Comment by ChuckHastings** [comment]
@gpucibot merge

**Pull request merged**